ext/ffi/ffi.c

```diff
@@ -4452,6 +4461,39 @@ ZEND_METHOD(FFI, isNull) /* {{{ */
 }
 /* }}} */
 
+ZEND_METHOD(FFI, hasSym) /* {{{ */
+{
+	zend_string *symbol;
+	zend_long symbol_kind;
+	zend_ffi_symbol *sym = NULL;
+	bool kind_is_null = 1;
+
+	ZEND_FFI_VALIDATE_API_RESTRICTION();
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_STR(symbol);
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG_OR_NULL(symbol_kind, kind_is_null)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if(!kind_is_null && (symbol_kind > ZEND_FFI_SYM_FUNC || symbol_kind < ZEND_FFI_SYM_TYPE)) {
+		zend_argument_value_error(2, "value error");
```

Comment:
The error message (`Argument #2 ($symbol_kind) value error`) would not be very helpful; we should note that a valid symbol kind needs to be passed:
```suggestion
		zend_argument_value_error(2, "must be a valid symbol kind");
```